src/synapse/azext_synapse/commands.py

```diff
@@ -60,14 +74,23 @@ def load_command_table(self, _):
         g.command('list', 'list_by_workspace')
         g.custom_command('create', 'create_sql_pool', supports_no_wait=True)
         g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.custom_command('update', 'update_sql_pool')
         g.command('pause', 'pause')
         g.command('resume', 'resume')
         g.wait_command('wait')
 
+    with self.command_group('synapse workspace firewall-rule', synapse_firewallrules_sdk,
+                            client_factory=cf_synapse_client_ipfirewallrules_factory) as g:
+        g.command('list', 'list_by_workspace')
+        g.show_command('show', 'get')
+        g.custom_command('create', 'create_firewall_rule', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.wait_command('wait')
+
     # Data Plane Commands
     # Spark batch opertions
-    with self.command_group('synapse spark batch', synapse_spark_batch_sdk, client_factory=cf_synapse_spark_batch) as g:
-        g.custom_command('create', 'create_spark_batch_job')
+    with self.command_group('synapse spark job', synapse_spark_batch_sdk, client_factory=cf_synapse_spark_batch) as g:
+        g.custom_command('submit', 'create_spark_batch_job')

```

Comment:
>submit [](start = 26, length = 6)

Usually CLI commands use "create" instead of "submit"

Reply:
we want to align with azure data lake anyalytics. There is already CLI command using submit.
`az dla job submit`

---
In reply to: [424972214](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424972214) [](ancestors = 424972214)

Reply:
I don't think `az dla job submit` is not a good example for your reference.

---
In reply to: [425000110](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r425000110) [](ancestors = 425000110,424972214)

Reply:
@wangdongwei@pku.edu.cn  please take a look here.

---
In reply to: [425009980](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r425009980) [](ancestors = 425009980,425000110,424972214)

Reply:
Hi @jsntcy , we would like to use the verb `submit` because it is more natural compared with `create` on the subject `job`. Another reason is that, for the DELETE API, we use the verb `cancel`. `submit` and `cancel` are more consistent. We can not use `delete` because you cannot actually delete a job.

But if `create` is strongly recommended, we are okay to change the verb to `create`.